Project7.md

```diff
@@ -172,12 +178,61 @@ It is also very useful to write comments for each line describing what the forma
 
 If you have an error in your program, Spark will print an exception (or many exceptions). You'll have to read carefully through the logging output to find a reference to what line or part of your program actually failed.
 
+
+Creating a Recommendation System
+--------------------------------
+
+With Spark's output of popular co-views, all that's left is figuring out how to store the information efficiently in a manner such that it would be easily accessible by a recommendation service. For this, we will create another table in our database with only two columns. The first column will contain item-id's, and the second column will contain variable length strings that will contain a comma separated "list" of all item-ids that were co-viewed 3 or more times with the item in the first column. This format is not difficult to implement and is a natural choice for a recommendation service, which need only have an item-id to retrieve from the database a list of "recommended" item-ids for that item. So an example recommendation table could look like this:
+
+```
+item_id  | recommended_items
+---------|------------------
+1        |  5, 7, 9,
+2        |  3, 4
+3        |  4, 2, 1, 5,
+```
+
+
+To populate and access this recommendations table, you will interface with the MySQL database directly with a [python MySQL client](https://github.com/PyMySQL/mysqlclient-python) rather than Django's ORM. Population of the table will occur during the Pyspark job. Within your Pyspark program, you will need to:
+
+1. Initiate a connection with the database using PyMySQL.
+2. Create a recommendations table with the appropriate column types if there is not already one, and insure sure it is empty.
+3. Make appropriate INSERT queries while iterating through co-view results after the ```collect``` call, making sure to account for items that already have rows in the table, as well as those that don't.
+
+The spark containers that will contain your Pyspark program don't come with everything needed for PyMySQL right off the bat, so some package installation must be taken care of. After 'docker-composing up' but before running your spark job for the first time, manually enter the two spark containers and run the following commands (either individually by hand, or in script form like below):
```

Comment:
This manual step reduces the value of using docker. It means we can't just 'docker-compose up' our system anywhere and have it work.